lib/child_process.js

```diff
@@ -161,15 +168,40 @@ function exec(command, options, callback) {
 }
 
 const customPromiseExecFunction = (orig) => {
-  return (...args) => {
+  return (file, options = {}) => {
     let resolve;
     let reject;
     const promise = new Promise((res, rej) => {
       resolve = res;
       reject = rej;
     });
 
-    promise.child = orig(...args, (err, stdout, stderr) => {
+    if (typeof options !== 'object') {
+      return Promise.reject(
```

Comment:
This should make use of the `PromiseReject` from primordials. Do a search for `PromiseReject` in `lib` for an example of how to do that.

Reply:
I'll do it, thanks 👍